### PR TITLE
Add project structure diagram

### LINKED ISSUE #23: Add project structure diagram to README
Related to #22, but making this a separate issue to keep work units small.

As pointed out by @tz33cu, a project structure diagram like this:

https://github.com/ashleve/lightning-hydra-template/tree/main#project-structure

would help new users of this repo get oriented to the project.

### PR DESCRIPTION
**Pull request by NoraLoose**
As discussed [here](https://github.com/leap-stc/ClimSim/pull/51#issuecomment-1687238000), this adds a file `ARCHITECTURE.md` to the repo root. 

Closes #23.

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## Project structure
+
```
Comment (cisaacstern) [inline ARCHITECTURE.md:2]:
````suggestion

```
````

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+├── website                   <- Documentation
+|
+├── .gitignore                <- List of files ignored by git
+|
+|── LICENSE                   <- License
+|
+├── setup.py                  <- File for installing climsim_utils as a package
+|
+└── README.md
+
```
Comment (cisaacstern) [inline ARCHITECTURE.md:38]:
````suggestion
```
````

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+│   ├── RPN                      <- Randomized Prior Network
+│   └── cVAE                     <- Conditional Variational Autoencoder
+│
+├── climsim_utils             <- Installable python package for data preprocessing and model evaluation
+|
+├── dataset_statistics        <- Precomputed statistics of input and output variables in dataset
+│
+├── demo_notebooks            <- Jupyter notebooks with examples of how to use the dataset for ML 
+│
+├── evaluation                <- Tools for evaluating ML models
```
Comment (cisaacstern) [inline ARCHITECTURE.md:20]:
```suggestion
├── evaluation                <- Jupyter notebooks for evaluating ML models
```

Just a nit, but I think this seems more descriptive.

**Review by cisaacstern — COMMENTED** [review]
Awesome, this is very helpful!

Only thing aside from the inline edits is maybe we want to mention this in README or website somewhere?

**Comment by NoraLoose** [comment]
@cisaacstern just pinging you to see what you think about the last commit. No rush, though!

**Comment by cisaacstern** [comment]
@NoraLoose thanks for the reminder, and sorry for the delay.

Yes, basically I think that looks good, my only question is can we get a GitHub link in the top nav of the website? Like:

Screen Shot 2023-08-22 at 4 21 43 PM


My concern with the code repo link being pushed deeper into the docs is that there is not am easily discoverable link back from the website to the GitHub repo.

**Comment by NoraLoose** [comment]
The repo button at the top is a good idea. (I am just realizing how much I am missing such a button on another project website that I'm managing!) Added via the latest commit.

**Comment by cisaacstern** [comment]
Awesome then IMO this is good to go! Thanks @NoraLoose !!

**Pull request merged**